extensions/blocks/contact-form/components/jetpack-newsletter-integration-settings.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+import { BaseControl, Button, ExternalLink } from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import { useCallback, useEffect, useState, useMemo } from '@wordpress/element';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { createBlock } from '@wordpress/blocks';
+
+const useInsertConsentBlock = () => {
+	const selectedBlock = useSelect( select => select( 'core/block-editor' ).getSelectedBlock(), [] );
+	const { insertBlock } = useDispatch( 'core/block-editor' );
+
+	const insertConsentBlock = useCallback( async () => {
+		const buttonBlockIndex = ( selectedBlock.innerBlocks ?? [] ).findIndex(
+			( { name } ) => name === 'jetpack/button'
+		);
+		if ( buttonBlockIndex === -1 ) {
+			return;
+		}
+
+		const newConsentBlock = await createBlock( 'jetpack/field-consent' );
+		await insertBlock( newConsentBlock, buttonBlockIndex, selectedBlock.clientId, false );
+	}, [ insertBlock, selectedBlock.clientId, selectedBlock.innerBlocks ] );
+
+	return { insertConsentBlock };
+};
+
+const NoConsentBlockSettings = () => {
+	const { insertConsentBlock } = useInsertConsentBlock();
+
+	const [ numberOfContacts, setNumberOfContacts ] = useState( 0 );
+
+	useEffect( () => {
+		apiFetch( { path: '/wp/v2/feedback?per_page=1', parse: false } )
+			.then( request => request.headers.get( 'X-WP-Total' ) )
+			.then( Number.parseInt )
+			.then( setNumberOfContacts );
+	}, [] );
+
+	return (
+		<>
+			{ numberOfContacts >= 10 ? (
+				<p>
+					{ sprintf(
+						__(
+							'You’ve already collected %d email contacts. Why not make sure you have permission to email them too?',
+							'jetpack'
+						),
+						numberOfContacts
+					) }
+				</p>
+			) : (
+				<p>
+					{ __(
+						'You’re already collecting email contacts. Why not make sure you have permission to email them too?',
+						'jetpack'
+					) }
+				</p>
+			) }
```

Comment:
Do you think we'd need different wording in those 2 cases? Wouldn't the more general wording be enough, without the numbers? This would also make the panel more performant since we would not need that extra query.